Email functionality / Username change

leong96 opened the pull request:
This commit implements the email functionality requested in Slack, as well as implements the username change feature. This should handle #90 and #134.

functions.py: Moved the change username function to functions.py, saves us a couple lines of code in views.

admin.php: hooked up the change username form to the backend.

allAdmins: Hooked up the create admin form to the backend.

allusers: Hooked up the invite provider form to the backend.

provider: hooked up the change email form to the backend.

views: Added all calls to send_email to send emails in the situations specified in Slack.

leong96 committed:
Email functionality / Username change

This commit implements the email functionality requested in Slack, as well as implements the username change feature.

functions.py: Moved the change username function to functions.py, saves us a couple lines of code in views.

admin.php: hooked up the change username form to the backend.

allAdmins: Hooked up the create admin form to the backend.

allusers: Hooked up the invite provider form to the backend.

provider: hooked up the change email form to the backend.

views: Added all calls to send_email to send emails in the situations specified in Slack.

leong96 committed:
Update views.py

Altered the way the organization name is changed from a SQL statement to changing the user object, which makes it so the name will update immidiately without a reload.

leong96 committed:
Merge branch 'master' into hleong-email

R-Eli-Levasseur commented on grasa_event_locator/views.py:
```diff
@@ -343,6 +347,7 @@ def denyEvent(request, eventID):
         if request.user.is_authenticated and request.user.userinfo.isAdmin and request.user.userinfo.isActive:
                 p = Program.objects.get(pk=eventID)
                 p.delete()
+                print(send_email([str(User.objects.get(pk=p.user_id.user_id))], "GRASA - Event Denied", "Your edited event has been denied. Contact GRASA for details."))
```
This says your edited event,  should just be your event

leong96 replied:
Altered in a subsequent commit.

R-Eli-Levasseur commented on grasa_event_locator/templates/provider.php:
```diff
@@ -38,7 +38,7 @@
                             {% csrf_token %}
                              <div class="input-group mb-3 changeNameInput">
-                                <input type="text" class="form-control" placeholder="{{ user.userinfo.org_name }}" aria-label="{{ user.userinfo.org_name }}" value="{{ user.userinfo.org_name }}" name="changename">
+                                <input type="text" name="changename" class="form-control" placeholder="{{ user.userinfo.org_name }}" aria-label="{{ user.userinfo.org_name }}" value="{{ user.userinfo.org_name }}" name="changename">
```
name="changename" is in here twice

leong96 replied:
Also corrected in a subsequent commit.

leong96 committed:
Update views.py

Remove the word edited from the email that would deny a new event, rather than an edited one.

leong96 committed:
Update provider.php

Removed duplicate "name="changename",
